packages/invoices/src/OutboundShipment/DetailView/ContentArea.tsx

```diff
@@ -38,13 +40,57 @@ const Expand: FC<{
   }
 };
 
+export const useHighlightPlaceholderRows = (
+  rows: OutboundLineFragment[] | OutboundItem[] | undefined
+) => {
+  const { setRowStyles } = useRowStyle();
+
+  useEffect(() => {
+    if (!rows) return;
+    const placeholders = [];
+
+    // This is a verbose .filter() on `rows` to find the placeholder lines.
+    // There is an issue with using `filter()` on a type which is
+    // A[] | B[]
+    // https://github.com/microsoft/TypeScript/issues/44373
+    for (const row of rows) {
+      if ('type' in row) {
+        if (row.type === InvoiceLineNodeType.UnallocatedStock) {
+          placeholders.push(row.id);
+        }
+      } else {
+        const hasPlaceholder = row.lines.some(
+          line => line.type === InvoiceLineNodeType.UnallocatedStock
+        );
+        if (hasPlaceholder) {
+          // Add both the OutboundItem and the individual lines, as
```

Comment:
good catch - that's pretty sweet

Reply:
Was wondering if the styles should be slightly different.. like a slightly different colours for the aggregate line.

(Though, the callback implementation would probably be better for setting multiple different styles 😛 )

Reply:
Also I wonder if it's a little troublesome having the mini table use the same context. I wonder if it should use it's own 🤔